microovn/client/client.go

```diff
@@ -134,28 +134,60 @@ func getOvsdbSchemaVersion(ctx context.Context, c *client.Client, dbSpec *ovnCmd
 
 // DisableService sends request to disable service with name as specified in
 // "serviceName" argument.
-func DisableService(ctx context.Context, c *client.Client, serviceName string) (types.WarningSet, error) {
+func DisableService(ctx context.Context, c *client.Client, serviceName string) (types.WarningSet, types.RegenerateEnvResponse, error) {
```

Comment:
Could we add the `WarningSet` to the `RegenerateEnvResponse`, to avoid returning 3 values?

Reply:
the warning set and the regenerateEnvResponse are quite distinct and i dont personally see much against returning three values
however if we were to merge them, id move regenerateEnvResponse into warning set

Reply:
> i dont personally see much against returning three values

It's more that I think that there's added value in returning less objects, especially if they correlate strongly. In this case you'd get single object that represents response, and contains:
* Overall success of the request
* Warnings that happened
* Errors that happened

It would make it easier for the receiver of this return value to work with it (+ you don't have to come up with two variable names to assign your function results to :laughing: ).

> however if we were to merge them, id move regenerateEnvResponse into warning set

Wouldn't the `WarningSet` be a natural subset of the overall `regenerateEnvResponse`?